crates/oxc_linter/src/plugin.rs

```diff
@@ -0,0 +1,76 @@
+use std::{env, fs, rc::Rc, sync::Arc};
+
+use oxc_diagnostics::miette::{miette, LabeledSpan};
+use oxc_semantic::Semantic;
+use trustfall::{execute_query, Schema, TryIntoStruct};
+
+use crate::{
+    context::LintContext,
+    lint_adapter::{InputQuery, LintAdapter},
+};
+
+#[derive(Debug, serde::Deserialize)]
+struct SpanInfo {
+    span_start: u64,
+    span_end: u64,
+}
+
+pub struct LinterPlugin {
+    rules: Vec<InputQuery>,
+    schema: Schema,
+}
+
+impl LinterPlugin {
+    pub fn new() -> Self {
+        let queries_path = env::var("OXC_PLUGIN").unwrap();
+        let rules = fs::read_dir(queries_path)
+            .unwrap()
+            .filter_map(std::result::Result::ok)
+            .filter(|dir_entry| dir_entry.path().is_file())
+            .filter(|f| f.path().as_os_str().to_str().unwrap().ends_with(".ron"))
+            .map(|f| fs::read_to_string(f.path()))
+            .map(std::result::Result::unwrap)
+            .map(|rule| ron::from_str::<InputQuery>(rule.as_str()).unwrap())
+            .collect::<Vec<_>>();
+
+        let schema_file = std::include_str!("schema.graphql");
+        // TODO: parse this at compile time
+        let schema = Schema::parse(schema_file).expect("schema file failed to parse");
```

Comment:
Unfortunately not at the moment. The problem is that we need the underlying GraphQL schema parser to be `const` first.

But if we were trying to push perf to the limit here (at the cost of some complexity and maintainability), we could skip the entire Trustfall compiler frontend and save the IR for each query. Then, this would just load and run the IR in the interpreter directly, no parsing or schema-checking required.

The problem with this is that the IR is explicitly unstable, so doing this safely would require something like a `build.rs` script to generate the IR for the appropriate Trustfall version the linter is currently using.

In my experience, parsing Trustfall queries has never been close to the bottleneck — it's always the data source that is the slowest part by far. If this is ever not the case, please consider it a perf bug and open an issue on Trustfall :)